nbclassic/notebookapp.py

```diff
@@ -201,7 +201,7 @@ def get_template(self, name):
             """Return the jinja template object for a given name"""
             return self.settings['notebook_jinja2_env'].get_template(name)
 
-        nbclassic.base.handlers.IPythonHandler.get_template = get_template
+        jupyter_server.base.handlers.JupyterHandler.get_template = get_template
```

Comment:
That is a good change

Reply:
It's not clear to me if we need this - does nbclassic still need to patch the _base_ handler's get_template, or only the handlers it serves? Previously, it only patched the _notebook_'s. get_template. Since these are served from this extension, maybe no patch is needed?

The `notebook_jinja2_env` must at least match the extension name - it will raise KeyError as it is currently, where the name is nbclassic, but the requested env is for the `notebook` extension (this extension's name in 0.3).

Reply:
Is there any issue to leave the get_template patch as it is now. We can revisit that later (it is difficult for me to justify the need of not atm, would need to take more time)

Sounds logical to rename `notebook_jinja2_env` to `nbclassic_jinja2_env` Is everything still working correctly with that change?

Reply:
> Is everything still working correctly with that change?

It's unclear to me what 'correctly' should mean here, but I don't _think_ so. Patching jupyter_server's get_template is something we haven't ever done before, so I'm not sure it's good to _start_ doing it now. I'm not sure what problem that would solve, but I can certainly imagine it _breaking_ things (i.e. uses of the _default_ jinja env won't work as they are overridden).

I believe this patch was defined to ensure that handlers inheriting from notebook's JupyterHandler would load from nbclassic's jinja env, set up by jupyter-server because notebook's jinja env was never instantiated. But that's not an issue jupyter-server's JupyterHandler has, and patching the base jupyter-server JupyterHandler makes the original jinja env inaccessible to all. So if/when the two differ, I think classes inheriting from jupyter_server.base.JupyterHandler should _not_ use the nbclassic env.

I'm not sure what exactly will happen for e.g. an extension Handler that still imports `notebook.base.JupyterHandler` after dealing with the notebook_shims, etc., or indeed what _should_ happen.